### ISSUE TITLE
System.Net.Security crash on arm64-Mono

### ISSUE THREAD

**Issue by karelz** [dotnet/runtime#75484]
Frequency in [last 30 days](https://runfo.azurewebsites.net/search/tests/?q=started%3A%7E30+definition%3Aruntime+name%3A%22system.net.security.tests+work+item%22) in Runfo as of 9/12:

Day | Run | Details
-- | -- | --
9/11 | Rolling run [11861](https://runfo.azurewebsites.net/view/build/?number=11861) | net7.0-Linux-Release-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open
9/10 | PR [11765](https://runfo.azurewebsites.net/view/build/?number=11765) | net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open
9/9 | PR [11335](https://runfo.azurewebsites.net/view/build/?number=11335) | net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open
9/9 | PR [10792](https://runfo.azurewebsites.net/view/build/?number=10792) | net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open
9/9 | PR [10264](https://runfo.azurewebsites.net/view/build/?number=10264) | net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open
9/8 | Rolling run [9270](https://runfo.azurewebsites.net/view/build/?number=9270) | net7.0-Linux-Release-**arm64-Mono**_release-(Debian.11.Arm64.Open)Ubuntu.1804.Armarch.Open
9/8 | PR [8767](https://runfo.azurewebsites.net/view/build/?number=8767) | net7.0-Linux-Debug-**arm64-Mono**_release-(Debian.11.Arm64.Open)Ubuntu.1804.Armarch.Open &<br/>net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open &<br/>net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open
9/7 | PR [8334](https://runfo.azurewebsites.net/view/build/?number=8334) | net7.0-Linux-Debug-**arm64-Mono**_release-(Debian.11.Arm64.Open)Ubuntu.1804.Armarch.Open
9/7 | PR [8314](https://runfo.azurewebsites.net/view/build/?number=8314) | net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open
9/7 | PR [8197](https://runfo.azurewebsites.net/view/build/?number=8197) | net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open &<br/>net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open
9/7 | PR [8024](https://runfo.azurewebsites.net/view/build/?number=8024) | net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open
9/7 | PR [6989](https://runfo.azurewebsites.net/view/build/?number=6989) | net7.0-Linux-Debug-**arm64-Mono**_release-(Debian.11.Arm64.Open)Ubuntu.1804.Armarch.Open
9/6 | PR [6677](https://runfo.azurewebsites.net/view/build/?number=6677) | net7.0-Linux-Debug-**arm64-Mono**_release-(Debian.11.Arm64.Open)Ubuntu.1804.Armarch.Open
8/26-9/6 |  | 25x Core Dumps on **arm64-Mono** - 1x Rolling run [1975580](https://runfo.azurewebsites.net/view/build/?number=1975580) (8/30)

Example of error ([Console log](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-7964053fabd4433ba3/System.Net.Security.Tests/1/console.10a48bc6.log?%3Fhelixlogtype%3Dresult)):
```
  Discovering: System.Net.Security.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Net.Security.Tests (found 297 of 487 test cases)
  Starting:    System.Net.Security.Tests (parallel test collections = on, max threads = 6)
    System.Net.Security.Tests.SslStreamEKUTest.SslStream_SelfSignedClientEKUClientAuth_Ok [SKIP]
      Condition(s) not met: "IsRootCertificateInstalled"
    System.Net.Security.Tests.SslStreamEKUTest.SslStream_ServerEKUClientAuth_Fails [SKIP]
      Condition(s) not met: "IsRootCertificateInstalled"
    System.Net.Security.Tests.SslStreamEKUTest.SslStream_ClientEKUServerAuth_Fails [SKIP]
      Condition(s) not met: "IsRootCertificateInstalled"
    System.Net.Security.Tests.SslStreamEKUTest.SslStream_NoEKUServerAuth_Ok [SKIP]
      Condition(s) not met: "IsRootCertificateInstalled"
    System.Net.Security.Tests.SslStreamEKUTest.SslStream_NoEKUClientAuth_Ok [SKIP]
      Condition(s) not met: "IsRootCertificateInstalled"
    System.Net.Security.Tests.SslStreamCertificateTrustTest.SslStream_SendCertificateTrust_ThrowsOnUnsupportedPlatform [SKIP]
      Condition(s) not met: "DoesNotSupportSendingCustomCANamesInTls"
Process terminated.
Error while reaping child. errno = 10

=================================================================
	Native Crash Reporting
=================================================================
Got a SIGABRT while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Native stacktrace:
=================================================================
	0x7fb7424440 - Unknown

=================================================================
	External Debugger Dump:
=================================================================
[New LWP 25]
[New LWP 26]
```

#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|0|

**Comment by karelz** [comment]
@wfurt @akoeplinger could it be related to #74795?

**Comment by wfurt** [comment]
Yes, this looks likely  https://github.com/dotnet/runtime/issues/74795  but we will need to fix it separately. It is somewhat sad that `Quic` creeps into test sets that do not really use it. 
It will be sucked in via reflection from certificate validation code. 
cc: @bartonjs for visibility.

**Comment by karelz** [comment]
Reopening to track backport to 7.0

**Comment by wfurt** [comment]
closed by  #75603

**Comment by karelz** [comment]
Fixed in main (8.0) in PR #75488 and in 7.0 (RC2) in PR #75603
